service/common/src/main/java/org/apache/polaris/service/catalog/iceberg/IcebergCatalog.java

```diff
@@ -1328,6 +1344,25 @@ public void doCommit(TableMetadata base, TableMetadata metadata) {
       String newLocation = writeNewMetadataIfRequired(base == null, metadata);
       String oldLocation = base == null ? null : base.metadataFileLocation();
 
+      // TODO: we should not need to do this hack, but there's no other way to modify
+      // metadataFileLocation / currentMetadataLocation
+      if (updateMetadataOnCommit) {
+        try {
+          Field tableMetadataField = TableMetadata.class.getDeclaredField("metadataFileLocation");
+          tableMetadataField.setAccessible(true);
```

Comment:
This should probably be done in a `static` init method.

Reply:
It needs to be done after we obtain the new metadata location from `writeNewMetadataIfRequired`

Reply:
Obtaining the `Field` only needs the `Class`.

Reply:
Oh, you mean that we should just obtain the fields in a static init block? Yes, we can do that. Let me make that change